app/models/editable_fields_collection.rb

```diff
@@ -1,15 +1,25 @@
 class EditableFieldsCollection
-  attr_accessor :collection
-  def initialize(collection)
-    @collection = to_hash(collection)
+  attr_reader :collection
+  def initialize(editables_config)
+    @collection = to_hash(editables_config)
   end
 
+  def [](key)
+    begin
+      collection.fetch(key)
+    rescue
+      collection.fetch("editable_field").fetch(key)
+    end
+  end
   private
 
-  def to_hash(editable_fields)
+  def to_hash(editables_config)
     hash = {}
-    editable_fields.each do |field|
-      hash[field.label] = ::DataEditable.new(field)
+    editables_config.each do |param, editables|
+      hash[editables.first.class.to_s.underscore] ||= {}
```

Comment:
This line is really fragile, Using the safe navigation operator would be a small improvement but this is not really robust anyway
```suggestion
      hash[editables&.first&.class&.to_s&.underscore] ||= {}
```

Reply:
you only need & before first cause everything in ruby is a class !